public/js/analytics/averageCmc.js

```diff
@@ -0,0 +1,76 @@
+function GetColorCat(colors) {
```

Comment:
Can you find a way to get all this code inside of webpack? That would involve putting it in the `src` directory instead and adding entry points in the `webpack.common.js` file for each analytic script. That way we would be able to avoid all the copying-and-pasting present throughout this PR. We'd also get minification and the ability to use node packages, all of which are good. I've been trying to eliminate all the code in the `public/js` dir and this goes in the opposite direction. Feel free to tag me on discord if you have questions. It would also make sure that this code is subject to eslint.